src/client.ts

```diff
@@ -525,109 +516,88 @@ export class Client {
-        return cb(err);
+      const cb = promisifyCb(resolve, reject, _cb);
+      const { decoders, decoderType } = opts;
+      const fwConstants = getFwVersionConst(this.fwVersion);
+      if (!fwConstants.maxDecoderBufSz) {
+        return cb('Please update Lattice firmware.');
+      }
+      const payload = Buffer.alloc(3 + fwConstants.maxDecoderBufSz);
+      const encDecoders = Buffer.from(rlpEncode(decoders));
+      if (encDecoders.length > fwConstants.maxDecoderBufSz) {
+        return cb('Too much data to make request. Please remove some decoders.');
       }
-      const payload = Buffer.alloc(abiPayload.length + 10);
-      // Let the firmware know how many defs are remaining *after this one*.
-      // If this is a positive number, firmware will send us a temporary code
-      // to bypass user authorization if the user has configured easy ABI loading.
-      payload.writeUInt16LE(defs.length, 0);
-      // If this is a follow-up request, we don't need to ask for user authorization
-      // if we use the correct temporary u64
-      if (nextCode !== null) nextCode.copy(payload, 2);
-      abiPayload.copy(payload, 10);
-      return this._request(payload, 'ADD_ABI_DEFS', (err, res, responseCode) => {
-        if (responseCode && responseCode !== responseCodes.RESP_SUCCESS)
+      payload.writeUInt8(decoderType, 0);
+      payload.writeUInt16LE(encDecoders.length, 1);
+      encDecoders.copy(payload, 3);
+      return this._request(payload, 'ADD_DECODERS', (err, res, responseCode) => {
+        if (responseCode && responseCode !== responseCodes.RESP_SUCCESS) {
           return cb('Error making request.');
-        else if (err) return cb(err);
-        const decrypted = this._handleEncResponse(res, decResLengths.addAbiDefs);
-        // Grab the 8 byte code to fast track our next request, if needed
-        nextCode = decrypted.data.slice(65, 73);
-        // No defs left? Return success
-        if (defs.length === 0) return cb(null);
-        // Add the next set
-        this.addAbiDefs(defs, cb, nextCode);
+        } else if (err) {
+          return cb(err);
+        }
+        this._handleEncResponse(res, decResLengths.empty);
+        return cb(null);
       });
     })
   }
 
   /**
-   * `getAbiRecords` fetches a set of ABI records saved on the Lattice. You can fetch
-   * sequential records based on a starting index and count.
+   * `getDecoders` fetches a set of decoders saved on the target Lattice.
    * @category Lattice
    * @returns The decrypted response.
    */
-  public getAbiRecords (opts: { n: number, startIdx: number, category: string }, _cb?: (err?: string, data?: GetAbiRecordsData) => void, fetched?: GetAbiRecordsData): Promise<{ err?: string, data?: GetAbiRecordsData }> {
+  public getDecoders (opts: { decoderType: number, n?: number, startIdx?: number, skipTotal?: boolean }, 
+    _cb?: (err?: string, data?: { decoders: Buffer[], total: number }) => void): Promise<{ err?: string, data?: Buffer[] }> 
```

Comment:
This return type doesn't match the values returned on line 600.

Reply:
[756bc00](https://github.com/GridPlus/gridplus-sdk/pull/353/commits/756bc001e62e4ba9fce75734645727253137ddc2)